httpx/models.py

```diff
@@ -205,7 +206,10 @@ def __eq__(self, other: typing.Any) -> bool:
         return isinstance(other, (URL, str)) and str(self) == str(other)
 
     def __str__(self) -> str:
-        return self._uri_reference.unsplit()
+        url = self._uri_reference.unsplit()
+        if self._uri_reference.userinfo:
+            url = url.replace(self._uri_reference.userinfo, "[secure]")
```

Comment:
We might have to reimplement `rfc3986.URIReference.unsplit()` here. Also we shouldn't obscure the username field, just the password field.